packages/cli/src/databases/migrations/sqlite/1717498465931-AddActivatedAtUserSetting.ts

```diff
@@ -0,0 +1,5 @@
+import { AddActivatedAtUserSetting1717498465931 as BaseMigration } from '../common/1717498465931-AddActivatedAtUserSetting';
+
+export class AddActivatedAtUserSetting1717498465931 extends BaseMigration {
+	transaction = false as const;
```

Comment:
We had some problems with this historically. Honestly don't remember. @netroy if you can please shed some light here 🙏🏽

Reply:
this is unfortunately badly named. 
`transaction = false` actually doesn't disable the transaction. [it actually takes over the control of the transaction](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/databases/utils/migrationHelpers.ts#L184-L188), so that we can [disable foreign-key constraints before the migration, and re-enable them after the migration](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/databases/utils/migrationHelpers.ts#L73-L84), because many table operations in the sqlite driver destroy the entire table and create a new table, which can cascade and delete unrelated tables.

We had to do this before we forked typeorm, because back then there was no other way to tell typeorm to disable foreign keys before creating the transaction.
Now that we have forked typeorm, I think we should:
1. update the sqlite driver to use native alter table operations instead of recreating tables where ever possible.
2. update the sqlite driver to reliable call [`beforeMigration`](https://github.com/n8n-io/typeorm/blob/master/src/driver/sqlite/SqliteQueryRunner.ts#L38) so that we don't need these hacks anymore. my last attempt at this https://github.com/typeorm/typeorm/pull/10147